elastic/kibana
# Hides Spaces actions in serverless saved objects management

Closes #169424

## Summary

Adds check against `SpacesApi.hasOnlyDefaultSpace` to, when true, bypass registering the `share-to-space` and `copy-to-space` actions and the `Spaces` column in the saved objects management UI.

### Manual Testing
- Start Elasticsearch with `yarn es serverless`
- Start Kibana with `yarn start --serverless security`
- Navigate to `Project Settings`->`Management`->`Content` -> `Saved Objects`
- Verify that the `Spaces` column does not appear in the Saved Objects table
- Verify that the `Actions` column does not contain either the `Copy-to-space` or `Share-to-space` actions for any object (will still contain `Inspect` and `View relationships`)
- Stop Kibana and Elasticsearch
- Start Elasticsearch with `yarn es snapshot --license=trial`
- Start Kibana with `yarn start`
- Add sample flight data to generate some saved objects
- Navigate to `Stack Management`->`Kibana`->`Saved Objects`
- Verify that the `Spaces` column does appear in the Saved Objects table
- Verify that the `Actions` column does contain both the `Copy-to-space` or `Share-to-space` actions for applicable object types (e.g. data views)

### Automated Testing 
- src/plugins/saved_objects_management/public/services/action_service.test.ts
- src/plugins/saved_objects_management/public/services/column_service.test.ts
